### PR TITLE
Update parser regex for splitting

### PR DESCRIPTION
**Pull request by shamshirz**
Added a test to catch strings that don't start with
carets to ensure we still add them

### CODE REVIEW

File: test/lib/markdown_parser_test.exs [hunk 1]
```diff
@@ -7,18 +7,25 @@ defmodule HonGraffitiPhoenix.MarkupParserTest do
...
   end
 
-  test "it returns the original string if there is no style" do
-    parsed = MarkupParser.parse(@no_markdown)
-    assert length(parsed) == 1
-    assert hd(parsed).body == @no_markdown
+  test "it parses strings that don't start with carets" do
+    parsed = MarkupParser.parse(@caret_not_first)
+    assert length(parsed) == 2
+    assert hd(parsed).body == "this is a "
```
Comment (samgqroberts) [inline test/lib/markdown_parser_test.exs:28]: solid test case. perhaps also a test case with two adjacent `^`s?

File: test/lib/markdown_parser_test.exs [hunk 2]
```diff
@@ -7,18 +7,25 @@ defmodule HonGraffitiPhoenix.MarkupParserTest do
   @no_markdown "thisIsSoFakn' BORING"
   @broken_markdown "this^Qmakes no ^09s sense!!"
   @invalid_markup_code "^j doesn't coorespond to a code"
+  @caret_not_first "this is a ^rrandom string"
```
Comment (samgqroberts) [inline test/lib/markdown_parser_test.exs]: nit: misspelling on line 9 "coorespond"

File: test/lib/markdown_parser_test.exs (hunk 1 above)
Comment (shamshirz) [reply test/lib/markdown_parser_test.exs:28]: Yes...this is totally an issue...should be dealt with eventually
Created a ticket!!! https://trello.com/c/EdxFAFxu/23-test-what-happens-in-hon-for-attached-cases

**Review by rjdellecese — APPROVED** [review]
Cool thx for fixing this :bug:

**Pull request merged**